hubble/executor/helper.py

```diff
@@ -0,0 +1,927 @@
+
+
+def get_rich_console():
+    """
+    Function to get jina rich default console.
+    :return: rich console
+    """
+    return Console(
+        force_terminal=True if 'PYCHARM_HOSTED' in os.environ else None,
+        color_system=None if 'JINA_LOG_NO_COLOR' in os.environ else 'auto',
+    )
+
+
+def retry(
+    num_retry: int = 3,
+    message: str = 'Calling {func_name} failed, retry attempt {attempt}/{num_retry}. Error: {error!r}',
+):
+    """
+    Retry calling a function again in case of an error.
+
+    :param num_retry: number of times to retry
+    :param message: message to log when error happened
+    :return: wrapper
+    """
+
+    def decorator(func):
+        @functools.wraps(func)
+        def wrapper(*args, **kwargs):
+            for i in range(num_retry):
+                try:
+                    return func(*args, **kwargs)
+                except Exception as e:
+                    default_logger.warning(
+                        message.format(
+                            func_name=func.__name__,
+                            attempt=i + 1,
+                            num_retry=num_retry,
+                            error=e,
+                        )
+                    )
+                    if i + 1 == num_retry:
+                        raise
+
+        return wrapper
+
+    return decorator
+
+
+def random_identity(use_uuid1: bool = False) -> str:
+    """
+    Generate random UUID.
+
+    ..note::
+        A MAC address or time-based ordering (UUID1) can afford increased database performance, since it's less work
+        to sort numbers closer-together than those distributed randomly (UUID4) (see here).
+
+        A second related issue, is that using UUID1 can be useful in debugging, even if origin data is lost or not
+        explicitly stored.
+
+    :param use_uuid1: use UUID1 instead of UUID4. This is the default Document ID generator.
+    :return: A random UUID.
+
+    """
+    return random_uuid(use_uuid1).hex
+
+
+def random_uuid(use_uuid1: bool = False) -> uuid.UUID:
+    """
+    Get a random UUID.
+
+    :param use_uuid1: Use UUID1 if True, else use UUID4.
+    :return: A random UUID.
+    """
+    return uuid.uuid1() if use_uuid1 else uuid.uuid4()
+
+
+def get_ci_vendor() -> Optional[str]:
+    with open(os.path.join(__resources_path__, 'ci-vendors.json')) as fp:
+        all_cis = json.load(fp)
+        for c in all_cis:
+            if isinstance(c['env'], str) and c['env'] in os.environ:
+                return c['constant']
+            elif isinstance(c['env'], dict):
+                for k, v in c['env'].items():
+                    if os.environ.get(k, None) == v:
+                        return c['constant']
+            elif isinstance(c['env'], list):
+                for k in c['env']:
+                    if k in os.environ:
+                        return c['constant']
+
+
+def get_full_version(jina_env: Optional[Dict] = None) -> Optional[Tuple[Dict, Dict]]:
```

Comment:
Why not passing version info to `HubIO` constructor? Will that be better?

Reply:
Receiving it in `HubIO` [constructor](https://github.com/jina-ai/jina-hubble-sdk/blob/feat-move-hubble-from-jina-core-to-here/hubble/executor/hubio.py#L63).

But [static methods](https://github.com/jina-ai/jina-hubble-sdk/blob/feat-move-hubble-from-jina-core-to-here/hubble/executor/hubio.py#L1044) using it too. So still need to pass this env values around.

Generally, I want to refactor the entire module, but maybe after copying it and having a working version first.